tsl/src/dist_util.c

```diff
@@ -113,6 +127,27 @@ dist_util_set_id(Datum dist_id)
 					 (errmsg("database is already a member of a distributed database"))));
 	}
 
+	if (check_metadata)
+	{
+		uuid = ts_metadata_get_value(CStringGetDatum(METADATA_UUID_KEY_NAME),
+									 CSTRINGOID,
+									 UUIDOID,
+									 &isnull);
+		if (isnull)
```

Comment:
I think this is wrong.  The `uuid` metadata is part of the telemetry and is automatically generated the first time we try to access it.  I'm not sure, but I think if a user has telemetry disabled, this might be expected to be null here, in which case we'd always hit this condition.

I think it is more correct to use ts_telemetry_metadata_get_uuid to get or generate the `uuid` field here, which has the added benefit of not creating a second definition of the METADATA_UUID_KEY_NAME in this file.

Reply:
Right. Switched to read the UUID using `ts_telemetry_metadata_get_uuid`.